tests/Unit/Parallel/Test_AlgorithmReduction.cpp

```diff
@@ -204,39 +157,87 @@ struct ArrayReduce {
     const auto& singleton_proxy = Parallel::get_parallel_component<
         SingletonParallelComponent<Metavariables>>(cache);
     /// [contribute_to_reduction_example]
+    Parallel::ReductionData<Parallel::ReductionDatum<int, funcl::Plus<>>>
+        my_send_int{array_index};
     Parallel::contribute_to_reduction<ProcessReducedSumOfInts>(
-        my_send_int, my_proxy, singleton_proxy, CkReduction::sum_int);
+        my_send_int, my_proxy, singleton_proxy);
     /// [contribute_to_reduction_example]
     /// [contribute_to_broadcast_reduction]
     Parallel::contribute_to_reduction<ProcessReducedSumOfInts>(
-        my_send_int, my_proxy, array_proxy, CkReduction::sum_int);
+        my_send_int, my_proxy, array_proxy);
     /// [contribute_to_broadcast_reduction]
 
-    double my_send_double = 13.4;
-    Parallel::contribute_to_reduction<ProcessReducedSumOfDoubles>(
-        my_send_double, my_proxy, singleton_proxy, CkReduction::sum_double);
-
-    Parallel::contribute_to_reduction<ProcessReducedProductOfDoubles>(
-        my_send_double, my_proxy, singleton_proxy, CkReduction::product_double);
+    /// [contribute_to_rms_reduction]
+    using RmsRed = Parallel::ReductionDatum<double, funcl::Plus<>,
```

Comment:
Yep this is better :) Can you make this a SquaredPlus as before, instead of squaring the arguments below?

Reply:
Because a0^2 + a1^2 + a2^2 != a0 + a1^2 + a2^2, and that's the difference between the two operations

Reply:
then I'm back at the original question: can we make a `Plus<Identity, Square<>>` work somehow? 😄

Reply:
Not without breaking `Plus<>`, no. The reason is the initial state, a0^2 on the lhs a0 on the rhs is the thing treated as `Identity` and there is no way around that. Since we don't know which of the reductions will be invoked in which order we have no idea which is the special initial state. Does that make sense?